packages/generator/src/v1.1/generateQueries.ts

```diff
@@ -0,0 +1,250 @@
+  importedTypes: Set<string>,
+): string {
+  switch (dataType.type) {
+    case "boolean":
+      return "boolean";
+    case "string":
+      return "string";
+    case "double":
+    case "float":
+    case "integer":
+    case "long":
+      return "number";
+
+    case "date":
+      importedTypes.add("LocalDate");
+      return "LocalDate";
+
+    case "timestamp":
+      importedTypes.add("Timestamp");
+      return "Timestamp";
+
+    case "attachment":
+      importedTypes.add("Attachment");
+      return "Attachment";
+
+    case "array":
+      return `Array<${
+        handleQueryDataType(dataType.subType, importedObjects, importedTypes)
+      }>`;
+
+    case "object": {
+      const objectType = dataType.objectTypeApiName!;
+      importedObjects.add(objectType);
+      return objectType;
+    }
+
+    case "objectSet": {
+      const objectType = dataType.objectTypeApiName!;
+      importedObjects.add(objectType);
+      importedTypes.add("ObjectSet");
+      return `ObjectSet<${objectType}>`;
+    }
+
+    case "set":
+      return `Set<${
+        handleQueryDataType(dataType.subType, importedObjects, importedTypes)
+      }>`;
+
+    case "struct":
+      const properties = dataType.fields.map(field => {
+        const isNullable = isNullableQueryDataType(field.fieldType);
+        return `${field.name}${isNullable ? "?" : ""}: ${
+          handleQueryDataType(field.fieldType, importedObjects, importedTypes)
+        }`;
+      });
+      return `{ ${properties.join(",\n")} }`;
+
+    case "union":
+      return dataType.unionTypes.map(type =>
+        handleQueryDataType(type, importedObjects, importedTypes)
+      ).join("|");
+
+    case "twoDimensionalAggregation":
+      importedTypes.add("TwoDimensionalAggregation");
+      dataType.valueType;
+      return `TwoDimensionalAggregation<
+        ${aggregationKeyToTypescriptType(dataType.keyType)},
+        ${aggregationValueToTypescriptType(dataType.valueType)}
+      >`;
+    case "threeDimensionalAggregation":
+      importedTypes.add("ThreeDimensionalAggregation");
+      return `ThreeDimensionalAggregation<
+        ${aggregationKeyToTypescriptType(dataType.keyType)},
+        ${aggregationKeyToTypescriptType(dataType.valueType.keyType)},
+        ${aggregationValueToTypescriptType(dataType.valueType.valueType)}
+      >`;
+
+    case "null":
+      return "null";
+
+    case "unsupported":
+      throw new Error("Cannot generate queries for unsupported type");
+
+    default:
+      const _: never = dataType;
+      throw new Error(
+        `Cannot generate queries for type ${(dataType as any).type}`,
+      );
+  }
+}
+
+function aggregationKeyToTypescriptType(
+  keyType: QueryAggregationKeyType,
+) {
+  switch (keyType.type) {
+    case "boolean":
+      return "boolean";
+    case "double":
+    case "integer":
+      return "number";
+    case "string":
+      return "string";
+    case "date":
+      return "LocalDate"; // TODO needs import at top of file
```

Comment:
Should we just import the default list and let prettier take care of unused imports?